src/components/elements/headers.tsx

```diff
@@ -11,7 +11,7 @@ export const H1 = ({children, className, ...props}: Props) => {
       {...props}
       className={twMerge(
         "rs-mb-5 font-medium",
-        className?.match(/type-\btext-\d+\b/g) ? className : ["type-3 xl:text-[4.1rem]", className]
+        className?.match(/(type-\b|\btext-\d+\b)/g) ? className : ["type-3 xl:text-[4.1rem]", className]
```

Comment:
Can you explain what regex your solving? I wonder if there's a more simple solution.

Reply:
The regex checks to see if the className contains either `type-` or `text-#`. This is so that `text-#` will be used in place of the default typography and to prevent it from passing with just `text-center` or `text-[color]`